pkgs/development/python-modules/imagededup/default.nix

```
@@ -0,0 +1,84 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchurl
+, cython
+, torch
+, torchvision
+, pillow
+, tqdm
+, scikit-learn
+, pywavelets
+, matplotlib
+, pytest
+, pytest-mock
+}:
+let
+  MobileNetV3 = fetchurl {
+    url = "https://download.pytorch.org/models/mobilenet_v3_small-047dcff4.pth";
+    hash = "sha256-BH3P9K3e+G6lvC7/E8lhTcEfR6sRYNCnGiXn25lPTh8=";
+  };
+  ViT = fetchurl {
+    url = "https://download.pytorch.org/models/vit_b_16_swag-9ac1b537.pth";
+    hash = "sha256-msG1N42ZJ71sg3TODNVX74Dhs/j7wYWd8zLE3J0P2CU=";
+  };
+  EfficientNet = fetchurl {
+    url = "https://download.pytorch.org/models/efficientnet_b4_rwightman-7eb33cd5.pth";
+    hash = "sha256-I6uLzVvb72GnpDuRrcrYH2Iv1/NvtJNaVpgo13iIxE4=";
+  };
+in
+buildPythonPackage rec {
+  pname = "imagededup";
+  version = "0.3.2";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "idealo";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-B2IuNMTZnzBi6IxrHBoMDsmIcqGQpznd/2f1XKo1Oa4=";
+  };
+
+  nativeBuildInputs = [
+    cython
+  ];
+
+  propagatedBuildInputs = [
+    torch
+    torchvision
+    pillow
+    tqdm
+    scikit-learn
+    pywavelets
+    matplotlib
+  ];
+
+  nativeCheckInputs = [ pytest pytest-mock ];
+
+  preCheck = ''
+    # checks fail with: error: [Errno 13] Permission denied: '/homeless-shelter'
+    export HOME=$TMPDIR
+
+    # checks with CNN are preloaded to avoid downloads in check-phase
+    mkdir -p /build/.cache/torch/hub/checkpoints/
+    ln -s ${MobileNetV3} /build/.cache/torch/hub/checkpoints/mobilenet_v3_small-047dcff4.pth
+    ln -s ${ViT} /build/.cache/torch/hub/checkpoints/vit_b_16_swag-9ac1b537.pth
+    ln -s ${EfficientNet} /build/.cache/torch/hub/checkpoints/efficientnet_b4_rwightman-7eb33cd5.pth
```

```suggestion
    ln -s ${MobileNetV3} $HOME/.cache/torch/hub/checkpoints/mobilenet_v3_small-047dcff4.pth
    ln -s ${ViT} $HOME/.cache/torch/hub/checkpoints/vit_b_16_swag-9ac1b537.pth
    ln -s ${EfficientNet} $HOME/.cache/torch/hub/checkpoints/efficientnet_b4_rwightman-7eb33cd5.pth
```
those hashes could be different with the next update, right?